src/components/accordion-disclosure/index.tsx

```diff
@@ -32,22 +63,25 @@ const AccordionDisclosure = ({
 				containerClassName={generateContainerClassName}
 				initialOpen={initialOpen}
 			>
-				<DisclosureActivator
-					className={s.button}
-					data-heap-track="accordion-disclosure-activator"
-				>
-					<span className={s.labelContainer}>
-						<Text asElement="span" className={s.title} weight="semibold">
-							{title}
-						</Text>
-						{description && (
-							<Text asElement="span" className={s.description} size={200}>
-								{description}
+				<DisclosureHeadingWrapper level={activatorHeadingLevel}>
```

Comment:
🚧 Just reiterating a comment I made in Slack earlier -- we should wait to add heading elements to this HTML until we can 100% ensure that the headings rendered to not break the heading outline of a page.

Reply:
For context, I think wrapping `button` in a heading element is necessary to match [the recommended Accordion pattern](https://www.w3.org/WAI/ARIA/apg/patterns/accordion/).

I think it might also be worth clarifying that with `DisclosureHeadingWrapper` as in this PR, `button` will be wrapped in a heading element _only if_ `activatorHeadingLevel` is provided. So for example on the [Boundary API docs pages](https://dev-portal-git-zsadd-accordion-component-hashicorp.vercel.app/boundary/api-docs/account-service), and anywhere else `AccordionDisclosure` might be used without an `activatorHeadingLevel` prop explicitly provided, `button` will _not_ be wrapped in a heading element - we'll retain the same outline as we have upstream on `main`.

I think we can 100% ensure that headings will not break the outline _unless_ we explicitly provide an `activatorHeadingLevel` that is incorrect. This might not be a proper 100% guarantee, but it's as close as I can imagine use getting in a pragmatic way.

Does that feel like an acceptable solution for now? Or would there be a change here that we feel like would decrease the chance of breaking heading outlines on a page?

Reply:
A heading element is not required to wrap the `AccordionDisclosure` `<button>`. I understand that the heading only renders with a certain prop, but I think adding this functionality before we can ensure that it doesn't break heading order in a page is taking on unnecessary tech debt that will impact us negatively later. We don't have anything automated in place to prevent broken heading orders, and the risk for breaking heading orders will increase when this component is eventually exposed as an authoring component. We should defer adding the ability to render a heading element until we can ensure accessibility won't be broken.

Reply:
Ah, alright! I was going off of this section of the [Accordion pattern](https://www.w3.org/WAI/ARIA/apg/patterns/accordion/):

> Each accordion header button is wrapped in an element with role [heading](https://w3c.github.io/aria/#heading) that has a value set for [aria-level](https://w3c.github.io/aria/#aria-level) that is appropriate for the information architecture of the page.

It looks like other parts of the `WAI-ARIA Roles, States, and Properties` are explicitly marked as optional, so my reading was that the heading wrapping around `button` was a required part of the accordion pattern. If our stance is that that's not the case, I can remove the heading-related code from this PR.

That being said, I don't really see how we can get a better guarantee on the semantics here - with this PR implementing the component only, and not using it anywhere, we're not exposing author control at all to the heading levels in use by the component. In Certifications work, I've been careful to make the heading level part of the code, that would have be changed in a PR in order to cause breakage.

Personally I think there's relatively low risk here to causing accessibility breakage; and implementing the heading logic seems like it might be a recommended part of the accordion pattern.

Reply:
✅ have updated to remove all `heading` related logic, per discussion thread in Slack.